youtube_dl/extractor/kommunetv.py

```diff
@@ -6,7 +6,8 @@
 
 
 class KommunetvIE(InfoExtractor):
-    _VALID_URL = r'https://(\w+).kommunetv.no/archive/(?P<id>\w+)'
+    _VALID_URL = r'https?://\w+\.kommun(?:etv\.no|\.tv)/(?:archive|live)/(?P<id>\w+)'
```

Comment:
```suggestion
    _VALID_URL = r'(?P<where>https?://\w+\.kommun(?:etv\.no|\.tv))/(?P<what>archive|live)/(?P<id>\w+)'
```
Then use the `where` group in the API URL (maybe Stavanger.kommunetv.no/api has diferent data from Oslo.kommunetv.no/api ?). Use the `what` to generate the correct `streamType` for the API, using `._download_json_handle(..., expected_status=404)` for the live case and then chasing down the archive content if the live is gone.